contracts/linear/src/events.rs

```diff
@@ -593,19 +598,36 @@ mod tests {
     }
 
     #[test]
-    fn validator_updated() {
+    fn validator_weight_updated() {
         let account_id = &alice();
-        let base_stake_amount = &U128(50000);
-        let weight: u16 = 10;
-        Event::ValidatorUpdated {
+        let old_weight: u16 = 10;
+        let new_weight: u16 = 20;
+        Event::ValidatorWeightUpdated {
+            account_id,
+            old_weight,
+            new_weight,
+        }
+        .emit();
+        assert_eq!(
+            test_utils::get_logs()[0],
+            r#"EVENT_JSON:{"standard":"linear","version":"1.0.0","event":"validator_weight_updated","data":[{"account_id":"alice","old_weight":10,"new_weight":20}]}"#
+        );
+    }
+
+    #[test]
+    fn validator_base_stake_amount_updated() {
+        let account_id = &alice();
+        let old_base_stake_amount = &U128(0);
+        let new_base_stake_amount = &U128(50000);
+        Event::ValidatorBaseStakeAmountUpdated {
             account_id,
-            base_stake_amount,
-            weight,
+            old_base_stake_amount,
+            new_base_stake_amount,
         }
         .emit();
         assert_eq!(
             test_utils::get_logs()[0],
-            r#"EVENT_JSON:{"standard":"linear","version":"1.0.0","event":"validator_updated","data":[{"account_id":"alice","base_stake_amount":"50000","weight":10}]}"#
+            r#"EVENT_JSON:{"standard":"linear","version":"1.0.0","event":"validator_base_stake_amount_updated","data":[{"account_id":"alice","old_base_stake_amount":"0","new_base_stake_amount":"50000"}]}"#
```

Comment:
We can make it easier to search in database.

What if
```rust
ValidatorUpdated {
    account_id: &'a AccountId,
    field: &`a String,  // or enum?
    old: &'a U128,
    new: &'a U128,
},
```
or change event names to `validator_updated_base_stake_amount` and `validator_updated_weight` so that we can search using sql:
```sql
select * from events where event like "validator_updated_%"
```

Reply:
Good point. However, we don't have a generic `events` table that we can query with SQL. 

We'll need to update subgraph mapping to make the data queryable, where we'll process the events respectively.

Reply:
Of course, but from the perspective of databases, searching with single prefix is quicker, same for Subgraph's underlying databases.

Reply:
Make sense. I don't see a strong requirement to search all the validator updated events, though I think it's a good idea to make it easier to index by aligning the prefixes. 

The schema of the subgraph table doesn't necessary follow the event names defined here. We'll probably have a `validator` entity that maintains its `prev_base_stake_amount` and `current_base_stake_amount` instead of storing all the events.

Reply:
Aligned prefixes may facilitate subgraph's optimization.

For example, if both weight and base amount events need to be read during an iteration, a subgraph search engine may be able to come up with a faster search method based on the same prefix.

If the underlying data of the subgraph is sharded (this is possible for older data), data with the same prefix is more likely to be divided into the same shard, reducing requests for data in the back end.

Of course, this is not critical.

Reply:
This is not how subgraph works. We don't index raw event data with subgraph.